www/app.py

```diff
@@ -15,6 +15,19 @@ def getblockdiagrams():
             names.append(xml.etree.ElementTree.parse(join('saved-bds', f)).getroot().find('designName').text)
     return jsonify(result=names)
 
+@app.route('/api/v1/blockdiagrams', methods=['POST'])
+def saveblockdiagram():
+    designName = request.form['designName']
+    bdString = request.form['bdString']
+    root = xml.etree.ElementTree.Element("root")
+
+    xml.etree.ElementTree.SubElement(root, 'designName').text = designName
+    xml.etree.ElementTree.SubElement(root, 'bd').text = bdString
+
+    tree = xml.etree.ElementTree.ElementTree(root)
+    tree.write('saved-bds/' + designName + '.xml')
+    return redirect(request.url)
```

Comment:
I think this is all you need. Its an empty response body with a 200 OK.
`return ('', 200)`